query-service-impl/src/main/java/org/hypertrace/core/query/service/promql/QueryRequestToPromqlConverter.java

```diff
@@ -0,0 +1,224 @@
+package org.hypertrace.core.query.service.promql;
+
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_AVG;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_MAX;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_MIN;
+import static org.hypertrace.core.query.service.QueryFunctionConstants.QUERY_FUNCTION_SUM;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.protobuf.ByteString;
+import java.time.Duration;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.codec.binary.Hex;
+import org.hypertrace.core.query.service.ExecutionContext;
+import org.hypertrace.core.query.service.QueryTimeRange;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Expression.ValueCase;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.LiteralConstant;
+import org.hypertrace.core.query.service.api.QueryRequest;
+import org.hypertrace.core.query.service.api.Value;
+import org.hypertrace.core.query.service.pinot.ViewDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class QueryRequestToPromqlConverter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(QueryRequestToPromqlConverter.class);
+
+  private final ViewDefinition viewDefinition;
+  private final Joiner joiner = Joiner.on(", ").skipNulls();
+
+  QueryRequestToPromqlConverter(
+      ViewDefinition viewDefinition) {
+    this.viewDefinition = viewDefinition;
+  }
+
+  /**
+   * 1. selection should be equal to group by list
```

Comment:
Add this comments too:

```
Note:
   * This will handle only instant aggregation query or time series aggregation query
   * Pl. see the selection handler for more details.
```